pkg/updatestrategy/aws_ec2.go

```diff
@@ -0,0 +1,245 @@
+package updatestrategy
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/zalando-incubator/cluster-lifecycle-manager/api"
+)
+
+// EC2NodePoolBackend defines a node pool consiting of EC2 instances
+// managed externally by some component e.g. Karpenter.
+type EC2NodePoolBackend struct {
+	ec2Client ec2iface.EC2API
+	clusterID string
+}
+
+// NewEC2NodePoolBackend initializes a new EC2NodePoolBackend for
+// the given clusterID and AWS session and.
+func NewEC2NodePoolBackend(clusterID string, sess *session.Session) *EC2NodePoolBackend {
+	return &EC2NodePoolBackend{
+		ec2Client: ec2.New(sess),
+		clusterID: clusterID,
+	}
+}
+
+// Get gets the EC2 instances matching to the node pool by looking at node pool
+// tag.
+// The node generation is set to 'current' for nodes with up-to-date
+// userData,ImageID and tags and 'outdated' for nodes with an outdated
+// configuration.
+func (n *EC2NodePoolBackend) Get(nodePool *api.NodePool) (*NodePool, error) {
+	instances, err := n.getInstances(nodePool)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list EC2 instances of the node pool: %w", err)
+	}
+
+	nodes := make([]*Node, 0)
+	for _, instance := range instances {
+		instanceID := aws.StringValue(instance.InstanceId)
+		node := &Node{
+			ProviderID:    fmt.Sprintf("aws:///%s/%s", aws.StringValue(instance.Placement.AvailabilityZone), instanceID),
+			FailureDomain: aws.StringValue(instance.Placement.AvailabilityZone),
+			Generation:    currentNodeGeneration,
+			// not used in clc logic
+			// Ready: true,
+		}
+
+		// TODO: resolve template version and compare that
+
+		params := &ec2.DescribeInstanceAttributeInput{
+			InstanceId: instance.InstanceId,
+			Attribute:  aws.String("userData"),
+		}
+
+		resp, err := n.ec2Client.DescribeInstanceAttributeWithContext(context.TODO(), params)
+		if err != nil {
+			return nil, err
+		}
+
+		currentInstanceConfig := &InstanceConfig{
+			UserData: aws.StringValue(resp.UserData.Value),
+			ImageID:  aws.StringValue(instance.ImageId),
+			Tags:     make(map[string]string, len(instance.Tags)),
+		}
+
+		for _, tag := range instance.Tags {
+			currentInstanceConfig.Tags[aws.StringValue(tag.Key)] = aws.StringValue(tag.Value)
+		}
+
+		ltParams := &ec2.DescribeLaunchTemplateVersionsInput{
+			LaunchTemplateName: aws.String(awsValidID(n.clusterID) + "-" + nodePool.Name),
```

Comment:
Does this still work/match after updating the kubernetes-on-aws PR to use the `AWSNodeTemplate` resource for creating the launch template?

Reply:
No, it does not work; you are right.
I think I can remove this part right?

Reply:
Having this removed basically means that nodes are always considered up to date and would not be rolled by CLC strategy.

Is this handled another way?